admin/desk/utils.py

```diff
@@ -0,0 +1,179 @@
+                    'emails': customer['emails'],
+                    'background': customer['background'],
+                    'company': customer['company'],
+                    'link': customer['_links']['self']
+                }
+        except DeskError:
+            raise
+        return customer_data
+
+    def create_customer(self, data):
+        """ Creates a customer based on the given data.
+            Documentation: http://dev.desk.com/API/customers/#create
+            Example URL: https://yoursite.desk.com/api/v2/customers
+            Example data:
+            {
+              "first_name": "Johnny",
+              "emails": [
+                {
+                  "type": "work",
+                  "value": "johnny@acme.com"
+                },
+                {
+                  "type": "other",
+                  "value": "johnny@other.com"
+                }
+              ],
+              "custom_fields": {
+                "level": "super"
+              }
+            }
+        """
+        customer_link = None
+        try:
+            customer_json = self.call_post('customers', data)
+            if not customer_json.get('errors'):
+                customer_link = customer_json['_links']['self']['href']
+        except DeskError:
+            pass
+        return customer_link
+
+    def create_case(self, data):
+        """ Creates a case based on the given data.
+        Documentation: http://dev.desk.com/API/cases/#create
+        Example URL: https://yoursite.desk.com/api/v2/cases
+        Example data:
+        {
+          "type": "email",
+          "subject": "Creating a case via the API",
+          "priority": 4,
+          "status": "open",
+          "labels": [
+            "Spam",
+            "Ignore"
+          ],
+          "language": "fr",
+          "message": {
+            "direction": "in",
+            "status": "received",
+            "to": "someone@desk.com",
+            "from": "someone-else@desk.com",
+            "cc": "alpha@desk.com",
+            "bcc": "beta@desk.com",
+            "subject": "Creating a case via the API",
+            "body": "Please assist me with this case",
+            "created_at": "2012-05-02T21:38:48Z"
+        }
+        """
+        case_link = None
+        try:
+            case_json = self.call_post('cases', data)
+            if not case_json.get('errors'):
+                case_link = case_json['_links']['self']['href']
+        except DeskError:
+            pass
+        return case_link
+
+    def cases(self, params):
+        case_list = [None]
+        params.update({
+            'sort_field': 'created_at',
+            'sort_direction': 'desc'
+        })
+        try:
+            case_list = self.call_get('cases/search', params)
+        except DeskError:
+            raise
+        return case_list[u'_embedded'][u'entries']
+
+    def create_find_customer_by_email(self, email, full_name=None):
```

Comment:
Minor: for consistency with other parts of the codebase, it would make sense to name this "find_or_create_customer_by_email".